src/lib/mqtt/MqttConnectionManager.cpp

```diff
@@ -43,6 +43,27 @@
 #define MQTT_DEFAULT_KEEPALIVE 60
 
 
+/* ****************************************************************************
+*
+* defaultTimeout -
+*/
+static long defaultTimeout = DEFAULT_TIMEOUT;
```

Comment:
Where this new variable is used?

(I see the new setMqttTimeout(), used in src/app/contextBroker/contextBroker.cpp, to set it, but I don't see in which places of the code it is actually used).

Reply:
In fact, we already have ` long timeout;` defined in src/lib/mqtt/MqttConnectionManager.h (L68). Is not a redundant setting? Why do we need this new one?

Reply:
Moreover, I don't see changes in `getConnection()` so it seems the new per-subscription timeout is no actually used in the MQTT broker connection logic.

Reply:
Moreover bis :), taking into account that we have a pool of MQTT connections and that several subscription (each one with a possibly different `timeout`), what happend if two subscription share the same connection? The first one to be triggered is the one which `timeout` is taken into account.

Let me illustrate with an examples. We have these MQTT subscriptions:

* Sub A for entity EA, timeout=10, MQTT endpoint E
* Sub B for entity EB, timeout=20, MQTT endpoint E

So:

* If EA is update before EB, then sub A is triggered first, so timeout 10 would be used
* If EB is update before EA, then sub B is triggered first, so timeout 20 would be used

In sum: depending of which subscription is triggered first, one different timeout may be used, which is pretty confusing for the user.